ipaserver/plugins/pkinit.py

```diff
@@ -121,6 +121,6 @@ def execute(self, *keys, **options):
         if server is not None:
             self.api.Object.server_role.ensure_master_exists(server)
 
-        result = sorted(self.get_pkinit_status(server, status))
+        result = self.get_pkinit_status(server, status)
```

Comment:
The issue seems rather to be that on ligne 96, pkinit_server_server key can be absent from ipa_master_config.

Reply:
This is fixing this error which happens when you call `sorted` on a type which does not implement comparisons:
```
2380] ipa: ERROR: non-public: TypeError: '<' not supported between instances of 'dict' and 'dict'
2380] Traceback (most recent call last):
2380]   File "/usr/lib/python3.6/site-packages/ipaserver/rpcserver.py", line 370, in wsgi_execute
2380]     result = command(*args, **options)
2380]   File "/usr/lib/python3.6/site-packages/ipalib/frontend.py", line 450, in __call__
2380]     return self.__do_call(*args, **options)
2380]   File "/usr/lib/python3.6/site-packages/ipalib/frontend.py", line 478, in __do_call
2380]     ret = self.run(*args, **options)
2380]   File "/usr/lib/python3.6/site-packages/ipalib/frontend.py", line 800, in run
2380]     return self.execute(*args, **options)
2380]   File "/usr/lib/python3.6/site-packages/ipaserver/plugins/pkinit.py", line 124, in execute
2380]     result = sorted(self.get_pkinit_status(server, status))
2380] TypeError: '<' not supported between instances of 'dict' and 'dict'
```

Example:
```
python3 -c "{'milan': 'jednicka'} > {'milan': 'dvojka'}"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
TypeError: '>' not supported between instances of 'dict' and 'dict'
```

Is there possibly one other error that would happen should the key was absent? Could it really be absent?

Reply:
Hi,
I had the issue with an IPA 4.4 server upgraded to 4.5:
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/ipaserver/rpcserver.py", line 370, in wsgi_execute
    result = command(*args, **options)
  File "/usr/lib/python3.6/site-packages/ipalib/frontend.py", line 450, in __call__
    return self.__do_call(*args, **options)
  File "/usr/lib/python3.6/site-packages/ipalib/frontend.py", line 478, in __do_call
    ret = self.run(*args, **options)
  File "/usr/lib/python3.6/site-packages/ipalib/frontend.py", line 800, in run
    return self.execute(*args, **options)
  File "/usr/lib/python3.6/site-packages/ipaserver/plugins/pkinit.py", line 124, in execute
    result = sorted(self.get_pkinit_status(server, status))
  File "/usr/lib/python3.6/site-packages/ipaserver/plugins/pkinit.py", line 96, in get_pkinit_status
    pkinit_servers = ipa_master_config['pkinit_server_server']
KeyError: 'pkinit_server_server'

This is a different error, though.